### PR TITLE
Romain/experiment notifications

### PR DESCRIPTION
**Pull request by romain-growthbook**
### Features and Changes

- Closes **(add link to issue here)**

### CODE REVIEW

File: packages/back-end/src/services/experimentNotifications.ts [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+
+  if (
+    !context.org.settings?.experimentNotificationsEnabled &&
+    !experiment.metricsNotificationsEnabled
+  )
+    return;
+
+  const [
+    {
+      results: [lastResult],
```
Comment (romain-growthbook) [inline packages/back-end/src/services/experimentNotifications.ts:217]: Get first result

File: packages/back-end/src/services/experimentNotifications.ts [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+  experiment: ExperimentInterface;
+  lastResult: ExperimentReportResultDimension;
+  snapshot: ExperimentSnapshotDocument;
+}) =>
+  updateWrapper({
+    context,
+    experiment,
+    type: "multiple-exposures",
+    handler: async () => {
+      const totalsUsers = lastResult.variations.reduce(
```
Comment (lukesonnet) [inline packages/back-end/src/services/experimentNotifications.ts:148]: Should we consider moving `multiExposureCheck` from the front-end into the shared package so the business logic lives in one place and use that method here?

File: packages/back-end/src/services/experimentNotifications.ts [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+}) =>
+  updateWrapper({
+    context,
+    experiment,
+    type: "srm",
+    handler: async () => {
+      const srmThreshold =
+        context.org.settings?.srmThreshold ?? DEFAULT_SRM_THRESHOLD;
+
+      if (srmThreshold <= lastResult.srm) return;
```
Comment (lukesonnet) [inline packages/back-end/src/services/experimentNotifications.ts:188]: Similarly I wonder if this should be in a shared package and used in both the FE and BE as well, akin to the `multiExposureCheck` method. This one is just a < comparison so maybe its an even dumber suggestion here, but I love the idea of the business logic being in one place.

File: packages/back-end/src/services/experimentNotifications.ts [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+      const totalsUsers = lastResult.variations.reduce(
+        (totalUsersCount, { users }) => totalUsersCount + users,
+        0
+      );
+      const percent = snapshot.multipleExposures / totalsUsers;
+      const multipleExposureMinPercent =
+        context.org.settings?.multipleExposureMinPercent ??
+        MINIMUM_MULTIPLE_EXPOSURES_PERCENT;
+
+      if (snapshot.multipleExposures < multipleExposureMinPercent) return;
```
Comment (lukesonnet) [inline packages/back-end/src/services/experimentNotifications.ts:157]: As we discussed in the call, may want to delete the old notification from `pastNotifications` if it exists and there is no longer multiple exposures issues.

**Review by lukesonnet — COMMENTED** [review]
minor comments on having business logic in `shared` package. Mostly just reviewed the ExperimentNotification.ts file.

File: packages/back-end/src/services/experimentNotifications.ts (hunk 1 above)
Comment (lukesonnet) [reply packages/back-end/src/services/experimentNotifications.ts:217]: from first analysis, and we also want to only fire these notifications for a certain set of snapshots (has no dimension might be the only check we do for now).

**Pull request closed without merging**